KeenClient/KeenClient.m

```diff
@@ -354,6 +354,11 @@ + (KeenClient *)sharedClientWithProjectID:(NSString *)projectID
         return nil;
     }
 
+    // If the keys are already set, return the existing instance, don't overwrite the keys
+    if (self.sharedClient.projectID || self.sharedClient.writeKey || self.sharedClient.readKey) {
+        return self.sharedClient;
```

Comment:
I think adding a log is a very good idea, I'll do that now. As for returning nil... I can see it going both ways. People might make this call with the same info, just because it's available? In that case the "correct" behavior feels like just returning the shared client?

I guess because this isn't explicitly an init method, it feels wrong to return nil?